packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -56,36 +62,87 @@ export const Skeleton = () => {
  * It also supports favourite products, controlled by a heart icon.
  */
 const ProductTile = (props) => {
-    const intl = useIntl()
     const {
-        product,
+        dynamicImageProps,
         enableFavourite = false,
+        imageViewType = 'large',
         isFavourite,
         onFavouriteToggle,
-        dynamicImageProps,
+        product,
+        selectableAttributeId = 'color',
         ...rest
     } = props
+    const {currency, imageGroups, price, productId, hitType} = product
+
+    const intl = useIntl()
+    const {buildUrl} = useMultiSite()
+    const {currency: activeCurrency} = useCurrency()
+    const isFavouriteLoading = useRef(false)
+    const styles = useMultiStyleConfig('ProductTile')
 
-    const {currency, image, price, productId, hitType} = product
+    const isMasterVariant = ['master', 'variant'].includes(hitType)
+    const initialVariationValue = isMasterVariant
+        ? product?.variants?.find((variant) => variant.productId == product.representedProduct.id)
+              ?.variationValues?.[selectableAttributeId]
+        : undefined
+
+    const [selectableAttributeValue, setSelectableAttributeValue] = useState(initialVariationValue)
```

Comment:
Question: I've been thinking about making this state and object instead of a single value. This might allow use some flexibility in the future as well as clean up some of the use's of that image utility that I created.